setup.sh

```diff
@@ -2,7 +2,79 @@
 # This script sets up the project by installing dependencies, checking for a poetry environment,
 # and installing pre-commit hooks.
 
+# Add color and formatting variables at the top
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+YELLOW='\033[1;33m'
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+BOLD='\033[1m'
 
+# Function for section headers
+print_section() {
+    echo -e "\n${BOLD}${BLUE}=== $1 ===${NC}\n"
+}
+
+# Function for success messages
+print_success() {
+    echo -e "${GREEN}✓ $1${NC}"
+}
+
+# Function for warnings
+print_warning() {
+    echo -e "${YELLOW}⚠ $1${NC}"
+}
+
+# Initialize and update git submodules
+print_section "Git Submodules"
+echo "Initializing git submodules..."
+git submodule update --init --recursive
+
+# ---- ENVIRONMENT VARIABLES ---- #
+# Source .env file if it exists
+print_section "Environment Variables"
+if [ -f .env ]; then
+    print_success "Loading environment variables from .env..."
+    source .env
+else
+    print_warning "No .env file found. You might need to create one with HF_TOKEN and WANDB_API_KEY"
+    echo -e "${YELLOW}Example .env contents:${NC}"
+    echo "export HF_TOKEN=your_huggingface_token"
+    echo "export WANDB_API_KEY=your_wandb_key"
+fi
+
+# ---- EVALUATION SETUP ---- #
+# Clone Paloma dataset if credentials are provided and directory doesn't exist
+print_section "Evaluation Setup"
+if [ ! -d "lib/paloma" ]; then
+    if [ ! -z "$HF_TOKEN" ]; then
+        echo "Cloning Paloma evaluation dataset..."
```

Comment:
Does not check if the credentials have access, returns successful even if the dataset is not downloaded.

```
=== Evaluation Setup ===

Cloning Paloma evaluation dataset...
Cloning into 'lib/paloma'...
remote: Access to dataset allenai/paloma is restricted and you are not in the authorized list. Visit https://huggingface.co/datasets/allenai/paloma to ask for access.
fatal: unable to access 'https://huggingface.co/datasets/allenai/paloma/': The requested URL returned error: 403
✓ Paloma dataset cloned successfully

```